src/lib/cache/URLDataCache.js

```diff
@@ -1,55 +1,58 @@
 // @flow
 import { globalFetchManager } from '../FetchManager';
 import { t } from 'ttag';
+
 import ResponseError from '../ResponseError';
+import TTLCache, { type TTLCacheOptions } from './TTLCache';
 
 // Provides a WhatWG-fetch-like API to make HTTP requests.
 // Caches response promises and returns an old promise if one is existing for the same URL.
 
+type URL = string;
+
 export default class URLDataCache<T> {
-  cache: { [key: string]: Promise<T> } = {};
+  cache: TTLCache<URL, Promise<T>>;
 
-  fetch(url: string, resolve: (data: T) => void, reject: (response: any) => void) {
-    this.constructor.fetch(url, { cordova: true }).then((response: Response) => {
+  constructor(options?: TTLCacheOptions) {
+    this.cache = new TTLCache<URL, Promise<T>>(options);
```

Comment:
0: This could happen directly in the declaration above

Reply:
How do you pass options into the cache then?